### PR TITLE
docs: Oryx boilerplate

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
## PR Description

Documentation on the boilerplate of Oryx. 

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

closes [HRZ-2151](https://spryker.atlassian.net/browse/HRZ-2151)

[HRZ-2151]: https://spryker.atlassian.net/browse/HRZ-2151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: _drafts/oryx/boilerplate.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+## How we prevent boilerplate code
+
+In Oryx, we use a few tactics to prevent boilerplate code. These tactics include:
+
+1. Bootstrap the application from npm packages rather than source.
+2. Expose an function to setup the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a [public repository](https://github.com/spryker/oryx) to enable developers to read the source code.
+5. Provide configurable component and business logic.
```
Comment (AndreyYevtukhov) [inline _drafts/oryx/boilerplate.md:21]: `configurable component` and `business logic` - should be links?

File: _drafts/oryx/boilerplate.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+6. Allow for customizations by [dependency injection](./dependency-injection.md).
+
+Most of the details around these tactics can be found elsewhere in the documentation. In the following sections you read more details on the boilerplate files that are needed.
+
+### NPM Packages
+
+One of the biggest sources of boilerplate code in any project is the application logic itself. In Oryx, we've separated out all the application logic into individual packages that are distributed on npm. These packages include components, business logic and integrations to the Spryker APIs.
```
Comment (AndreyYevtukhov) [inline _drafts/oryx/boilerplate.md:28]: `distributed on npm` - maybe also a link?

File: _drafts/oryx/boilerplate.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+By separating out the application logic in this way, we've eliminated the boilerplate code from your projects. Instead of writing code to handle basic functionality, you can simply install the appropriate package and use it directly in your code. And because the packages are distributed on npm and published with [Semantic Versioning](https://semver.org/), you can be confident that upgrading to a new version of the framework won't break your code.
+
+### Using Presets
+
+Another source of boilerplate code in any project is the configuration required to get your application up and running. To simplify this process, we've introduced the concept of presets in Oryx.
+
+Presets provide configurations and data structures that are designed to get you up and running quickly without providing a lot of configuration yourself. Presets can be considered as so-called "demo applications", as they typically represent a demo application for a specific business model (e.g. "b2c demo shop"). By using presets, you can quickly configure your application without writing a lot of boilerplate code. This is particularly great when you first start your project or want to demo or test things out.
```
Comment (AndreyYevtukhov) [inline _drafts/oryx/boilerplate.md:36]: Maybe a link to the `preset` folder in the `oryx` repo - so that devs can check it out?

File: _drafts/oryx/boilerplate.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+### Using appBuilder
```
Comment (AndreyYevtukhov) [inline _drafts/oryx/boilerplate.md:38]: Too many `boilerplate code` notes in this block (as for me)
Maybe rephrase a bit :)

Comment (AndreyYevtukhov) [reply _drafts/oryx/boilerplate.md:38]: Screenshot 2023-03-16 at 10 07 39

File: _drafts/oryx/boilerplate.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+- `server/server.ts` – This file is responsible for setting up the server and handling incoming requests.
+
+If you use vite, you can add an npm script in your package.json to build and serve the SSR server:
+
+```json
+"scripts": {
+  "build:ssr": "vite build -c vite.config.server.ts",
+  "serve:ssr": "ts-node --experimentalSpecifierResolution node --esm server/server",
+}
+```
```
Comment (AndreyYevtukhov) [inline _drafts/oryx/boilerplate.md:139]: maybe some "final words"?

File: _drafts/oryx/boilerplate.md (hunk 1 above)
Comment (tobi-or-not-tobi) [reply _drafts/oryx/boilerplate.md:21]: once we have those docs, yes.

File: _drafts/oryx/boilerplate.md (hunk 3 above)
Comment (tobi-or-not-tobi) [reply _drafts/oryx/boilerplate.md:36]: I'll make it a link to the preset package instead. We can consider making deeplinks from the preset package readme to the github code. But that's beyond this doc.

File: _drafts/oryx/boilerplate.md (hunk 4 above)
Comment (tobi-or-not-tobi) [reply _drafts/oryx/boilerplate.md:38]: have done a few tweaks but happy to leave it with Andrii to take it to next level

File: _drafts/oryx/boilerplate.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+Here's a breakdown of the different bits of boilerplate code:
+
+### `package.json`
+
+The `package.json` file contains all the dependencies of the project. To simplify the dependency management, all dependencies are pulled through a single preset package. The preset package contains dependencies to all available oryx packages. This might not be the most optimal setup for your project over time, as it might contain a lot of "dead code", but for the start of your project it is very convenient. While you're getting experienced with Oryx over time, you could consider to create a narrowed down list of dependencies.
+
+That being said, having unused dependencies in your project will _not_ affect the build time or run time of your project. It is only an overhead during the installation process.
+
+The bare minimum package.json is setup with the following package.json:
```
Comment (andriitserkovnyi) [inline _drafts/oryx/boilerplate.md:68]:
```suggestion
The bare minimum package.json is setup with the following dependencies:
```

File: _drafts/oryx/boilerplate.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+## How we prevent boilerplate code
+
+In Oryx, we use a few tactics to prevent boilerplate code. These tactics include:
+
+1. Bootstrap the application from [npm packages](https://www.npmjs.com/org/spryker-oryx) rather than source.
+2. Expose an function to setup the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a [public repository](https://github.com/spryker/oryx) to enable developers to read the source code.
+5. Provide configurable component and business logic.
```
Comment (andriitserkovnyi) [inline _drafts/oryx/boilerplate.md:21]:
```suggestion
5. Provide configurable components and business logic.
```

File: _drafts/oryx/boilerplate.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+In Oryx, we use the following tactics to prevent boilerplate code:
+
+1. Bootstrap the application from [npm packages](https://www.npmjs.com/org/spryker-oryx) instead of source.
+2. Expose a function to set up the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a [public repository](https://github.com/spryker/oryx) to enable developers to read the source code.
+5. Provide configurable components and business logic.
+6. Allow for customizations with [dependency injection](./dependency-injection.md).
+
+Most of the details around these tactics can be found elsewhere in the documentation. In the following sections you read more details on the boilerplate files that are needed.
```
Comment (andriitserkovnyi) [inline _drafts/oryx/boilerplate.md:27]: What are the files needed for?

Comment (tobi-or-not-tobi) [reply _drafts/oryx/boilerplate.md:27]: These files are the boilerplate. What about:

> In the following sections you read more details on the files that are part of the boilerplate.

**Review by VadymSachenko — CHANGES_REQUESTED** [review]
See comments.

**Review by VadymSachenko — APPROVED** [review]
